### PR TITLE
Replace Travis CI with Github actions CI

### PR DESCRIPTION
**Pull request by ps37**
### What:
Introduce Github actions as CI platform and add jobs to publish components library with @latest tag and @next tag.

### Publishing to @next:
Steps for publishing an @next release:
- Update version number in projects/lib-name/package.json
- Push the changes to a remote topic branch and create a pull request into vmware/vmware-cloud-director-ui-components/master
- Upon approval of the PR, merge that PR into master

Following the above steps makes the CI-CD pipeline to execute publishing job to npm with @next tag(npm publish --tag next)

Test run: https://github.com/ps37/vmware-cloud-director-ui-components/runs/517873277?check_suite_focus=true

### Publishing to @latest:
Steps for publishing an @latest release:
- Update version number in projects/lib-name/package.json
- Push the changes to a remote topic branch and create a pull request into vmware/vmware-cloud-director-ui-components/master
- Upon approval of the PR, Push the changes to remote repo using Git tag using following commands:
    - Add a Git tag to the HEAD commit that has to be published as latest: `git tag -fa <lib-name>-v[0-999].[0-999].[0-999]`
    - Push to the remote repo(vmware/vmware-cloud-director-ui-components): `git push git@github.com:vmware/vmware-cloud-director-ui-components.git refs/tags/<lib-name>-v[0-999].[0-999].[0-999]`

Following the above steps makes the CI-CD pipeline to execute publishing job to npm with @next tag(npm publish --tag next)

Test run: https://github.com/ps37/vmware-cloud-director-ui-components/runs/519800438?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+              uses: ./.github/actions/check-lib-package
+              with:
+                  lib-name: 'components'
+            - name: Publish components@next
+              if: steps.check-components-tag.outputs.isLibTag != 'true' && steps.check-components-package.outputs.isLibPackageChanged == 'true'
+              run: |
+                  cd ./dist/components
+                  npm publish
+              env:
+                  NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
```
Comment (ryan-bradford) [inline .github/workflows/ci-cd.yml]: Can we add publishing for i18n for latest and next?

File: projects/components/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-    "name": "@vcd/ui-components",
-    "version": "0.0.5",
+    "name": "@ps37/ui-components",
+    "version": "0.13.0",
```
Comment (ryan-bradford) [inline projects/components/package.json]: Revert these changes and the ones in angular.json?

File: .github/actions/check-lib-tag/index.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const core = require('@actions/core');
+const github = require('@actions/github');
+
+try {
+    const libName = core.getInput('lib-name');
+    const refStringArray = github.context.ref.split('/');
```
Comment (juanmendes) [inline .github/actions/check-lib-tag/index.js]: Let's make it pop

```
// Show sample value of github.context.ref
const branchName = github.context.ref.split('/').pop()
```

File: README.md [hunk 4]
```diff
@@ -60,7 +60,7 @@ the extremely long folder name.
 
 ## Publishing
 
-Publishing happens through the CI/CD pipeline. See [package.json](.travis.yml)
+Publishing happens through the CI/CD pipeline. See [package.json](.github/workflows/ci-cd.yml)
```
Comment (juanmendes) [inline README.md]: We should have an explanatino instead of just saying CI/CD. Two sections, creating an `@next` release and creating an `@latest`

File: package.json [hunk 5]
```diff
@@ -29,6 +29,8 @@
   },
   "private": true,
   "dependencies": {
+    "@actions/core": "^1.2.3",
```
Comment (juanmendes) [inline package.json]: Should be dev deps and should have exact numbers. Remember to always use `npm i  lib-name --save-exact`

File: .github/workflows/ci-cd.yml [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+                  lib-name: 'components'
+            - name: Publish components@next
+              if: steps.check-components-tag.outputs.isLibTag != 'true' && steps.check-components-package.outputs.isLibPackageChanged == 'true' && github.event_name == 'push'
+              run: |
+                  cd ./dist/components
+                  npm publish
+              env:
+                  NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+
+    publish-doc-lib-latest:
```
[the remark this answers is not in the document]
Comment (ps37) [reply .github/workflows/ci-cd.yml]: https://jira.eng.vmware.com/browse/VDUCC-123

File: README.md [hunk 7]
```diff
@@ -60,7 +60,24 @@ the extremely long folder name.
...
+Steps for publishing an @next release:
```
Comment (juanmendes) [inline README.md]: This needs some sort of header
### Steps for publishing... 

Same for `@latest`

**Comment by juanmendes** [comment]
Please comment out tests clipped text directive tests and we'll fix it in https://jira.eng.vmware.com/browse/VDUCC-116

File: .github/workflows/ci-cd.yml [hunk 8]
```diff
@@ -0,0 +1,369 @@
...
+            - uses: actions/cache@v1
+              with:
+                  path: ./coverage
+                  key: coverage
+            - name: Run unit-tests
+              run: |
+                  npm run test:ci:components
+                  npm run test:ci:doc-lib
+                  npm run test:ci:i18n
+            - name: Upload coverage to Codecov
```
Comment (ryan-bradford) [inline .github/workflows/ci-cd.yml]: Does this upload the Codecov report, and if so, why do we need to keep the .travis.yml? And if we are keeping it and requiring the build to run to post coverage results, doesn't that reduce the benefit of switching over to GH Actions?
